aptos-core/consensus/src/block_storage/sync_manager.rs

```diff
@@ -536,33 +521,77 @@ impl BlockStore {
         info!("process_block_retrieval origin_block_id {}, target_block_id {}",
                 request.req.block_id(), request.req.target_block_id().unwrap());
         let mut blocks = vec![];
+        let mut quorum_certs = vec![];
         let mut status = BlockRetrievalStatus::Succeeded;
-        let mut id = request.req.block_id();
+        let (retrieval_epoch, mut id) = if let Some(epoch) = request.req.epoch() {
+            let target_block_number = self
+                .storage
+                .consensus_db()
+                .get_all::<EpochByBlockNumberSchema>()
+                .unwrap()
+                .into_iter()
+                .find(|(_, eppch_)| *eppch_ == epoch)
```

Comment:
typo: eppch_ => epoch

Reply:
This is correct. It only requires data of the same epoch.